src/diffusers/utils/doc_utils.py

```diff
@@ -36,3 +36,41 @@ def docstring_decorator(fn):
         return fn
 
     return docstring_decorator
+
+
+def add_start_docstrings(*docstr):
+    def docstring_decorator(fn):
+        fn.__doc__ = "".join(docstr) + (fn.__doc__ if fn.__doc__ is not None else "")
+        return fn
+
+    return docstring_decorator
+
+
+def add_start_docstrings_to_model_forward(*docstr):
+    def docstring_decorator(fn):
+        docstring = "".join(docstr) + (fn.__doc__ if fn.__doc__ is not None else "")
+        class_name = f"[`{fn.__qualname__.split('.')[0]}`]"
+        intro = f"   The {class_name} forward method, overrides the `__call__` special method."
+        note = r"""
+
+            <Tip>
+
+            Although the recipe for forward pass needs to be defined within this function, one should call the
+            [`Module`] instance afterwards instead of this since the former takes care of running the pre and post
+            processing steps while the latter silently ignores them.
```

Comment:
I don't think all the modules need to have this. Pipelines are not `nn.Module`.

Reply:
I think the utils here have been copied over from `transformers`. 

example usage:
https://github.com/huggingface/transformers/blob/831bc25d8fdb85768402f772cf65cc3d7872b211/src/transformers/models/fnet/modeling_fnet.py#L530

We could look into adding docstrings this way for the model components? It's not necessary though.

Reply:
Yeah we certainly don't need that. Let's remove this then?